src/generic_ui/scripts/content_digitalocean.js

```diff
@@ -1,35 +1,54 @@
-(function () {
+(function improveDOWelcomeFlow() {
+  // TODO: check if cloud server setup has ever been completed successfully,
+  // and only proceed if it hasn't.
+
+  var baseUrlFF = '';  // base url for local uProxy assets in Firefox
+  if (typeof self !== 'undefined' && self.port && self.port.on) {
+    self.port.on('baseUrlFF', function (baseUrl) {
+      baseUrlFF = baseUrl;
+    });
+    var getLocalAssetUrl = function (relativeUrl) {
+      return baseUrlFF + relativeUrl;
+    };
+  } else if (typeof chrome !== 'undefined' && chrome.extension && chrome.extension.getURL) {
+    var getLocalAssetUrl = chrome.extension.getURL;
+  } else {
+    var getLocalAssetUrl = function (relativeUrl) {
+      // Fall back to GitHub
+      return 'https://rawgit.com/uProxy/uproxy/master/src/' + relativeUrl;
+    };
+  }
+
   var pageUrl = document.location.href,
       pageId = getPageId(pageUrl),
       pmntAdded = isPmntAdded();
 
-  console.log('pageId:', pageId);
-  console.log('pmntAdded:', pmntAdded);
-
   if (pageId === 'welcome' && pmntAdded) {
     if (shouldPromptPromo()) {
+      // We need to redirect to the billing page to use its promo form.
       document.location.href = 'https://cloud.digitalocean.com/settings/billing';
     }
   }
   if (pageId === 'billing' && pmntAdded) {
-    showOverlay();
+    // Delay execution by a bit, so baseUrlFF has a chance to get set.
+    setTimeout(showOverlay, 500);
```

Comment:
OK If I do this in a follow-up PR, so we can get this landed and I can work on some higher-priority stuff for OFF first?